pallets/nft-manager/src/tests/single_mint_nft_tests.rs

```diff
@@ -169,6 +169,28 @@ mod mint_single_nft {
             });
         }
 
+        #[test]
+        fn external_ref_is_out_of_bounds() {
+            let mut ext = ExtBuilder::build_default().as_externality();
+            ext.execute_with(|| {
+                let mut context: Context = Default::default();
+                let expected_info_id: NftInfoId = NftManager::next_info_id();
+
+                assert_eq!(false, <Nfts<TestRuntime>>::contains_key(&context.generate_nft_id()));
+                assert_eq!(false, <NftInfos<TestRuntime>>::contains_key(&expected_info_id));
+                assert_eq!(false, context.event_emitted_with_single_nft_minted());
+
+                let out_of_bounds_size = 4096;
+                context.unique_external_ref = (0..out_of_bounds_size).map(|_| 65).collect();
+
```

Comment:
I can change this to use 1025 instead of 4096 to include the edge case. Any value outside the boundaries tests the boundary itself, let's not forget that.
Refactored it a bit to be easier to read, as I would like to avoid creating a function for 2 lines that initialise a variable to a value. It disrupts the flow when reading the code when you have to jump to a different area of the codebase to see that function implementation.

Reply:
I agree with you. It does disrupt the flow. But that is where good naming comes in. You should not have to jump to the implementation when you first meet the function, you should trust it does what it says and carry on. That is where function extraction and good naming help in reading the code, you stay in the same place (as you would without extracting the function) and you stay at the same abstraction level (which is the improvement over _not extracting the function_).

Then, when you review the rest of the code, you'll eventually come across the new function. There, you just have to check two things: that the name properly corresponds to the implementation. Again, no jumping around: it's merely about checking the contract between functions is well-defined and well implemented.

As for the testing changes. I agree, any value outside the boundaries tests the changes, so with an infinity of values, which one do you choose? I try to choose _the first_ that triggers the behaviour I want to test. There is some trust, in here, in the monotonicity of the condition: if it triggers for value _n_, it will also trigger for value _n+1_. So, if the test is made for the first value that should trigger it, I believe that the behaviour will be correct for all the necessary cases. If I only do it for the 2000th case, then I have no tests for the space in between.

Ideally, what I like to do in my tests, is even more complete: test both sides of the boundary. If a condition should trigger on index _n_, I write two cases: 
* behaviour triggers on index _n_
* behaviour does not trigger on index _n-1_
Again, monotonicity is assumed.